### PR TITLE
mockGCP: Mock ComputeFirewallPolicy and mock ComputeFirewallPolicyRule

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

Commit 1: collect logs from realGCP
Commit 2: add mocks for ComputeFirewallPolicy and compare mock logs with realGCP
Commit 3: add mocks for ComputeFirewallPolicyRule and compare mock logs with realGCP

I'm still seeing some diffs with realGCP. Comment in the PR and we can determine if they are acceptable.

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log
Comment (gemmahou) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log:33]: I think targetId and targetLink are necessary in mock compute operation?

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrulefull/_http.log
Comment (gemmahou) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrulefull/_http.log]: `routingConfig.bgpBestPathSelectionMode` is a field belongs to ComputeNetwork. Per the [doc](https://cloud.google.com/compute/docs/reference/rest/beta/networks), this field is `The BGP best path selection algorithm to be employed within this network for dynamic routes learned by Cloud Routers. Can be LEGACY (default) or STANDARD.` However, I do not see this field in the [proto](https://raw.githubusercontent.com/googleapis/googleapis/refs/heads/master/google/cloud/compute/v1/compute.proto)...

I feel it's fine to skip this diff, since network is just a reference of the policy rule re source.

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrulefull/_http.log
Comment (gemmahou) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrulefull/_http.log]: This one I think we need to fix, this error comes from getRule method, but I'm unsure why the "errors" field is missing...

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrule/_http.log
Comment (gemmahou) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrule/_http.log]: This one I think we need to fix, this error comes from getRule method, but I'm unsure why the "errors" field is missing...

File: mockgcp/mockcompute/firewallpoliciesv1.go
Comment (justinsb) [inline mockgcp/mockcompute/firewallpoliciesv1.go:200]: Nit: You can just use time.Now and rely on normalization, we have to normalize anyway for real-gcp tests

File: mockgcp/mockcompute/firewallpoliciesv1.go
Comment (justinsb) [inline mockgcp/mockcompute/firewallpoliciesv1.go:247]: Nit: refactor to common function (e.g. PopulateDefaultsForRule)

File: mockgcp/mockcompute/firewallpoliciesv1.go
Comment (justinsb) [inline mockgcp/mockcompute/firewallpoliciesv1.go:317]: Updating in a GET is a bit suspicious.  Do you need to do this for some reason?  If so, I think a comment would be helpful!

File: mockgcp/mockcompute/firewallpoliciesv1.go
Comment (justinsb) [inline mockgcp/mockcompute/firewallpoliciesv1.go:396]: Nit: do you think we should check whether the rule was found?  Maybe not, because we don't have any examples of it so we don't know what to do (and don't need it!)

File: mockgcp/mockcompute/firewallpoliciesv1.go
Comment (justinsb) [inline mockgcp/mockcompute/firewallpoliciesv1.go:402]: Nit: consider refactoring into a shared helper (along with setting Kind and Description, maybe)

File: mockgcp/mockcompute/firewallpoliciesv1.go
Comment (justinsb) [inline mockgcp/mockcompute/firewallpoliciesv1.go:485]: Nit: You should probably check tokens[0] and tokens[1] also, just defensive coding

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log
Comment (justinsb) [inline pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log:1]: I like that you recorded the real GCP log first and committed it - it makes it easy to see the diffs 👍

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log:33]: Not sure, but whatever the real-gcp does, that's what we want to simulate.

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrule/_http.log
Comment (justinsb) [reply pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicyrule/_http.log]: Populating the detailed errors is pretty tough in general.  Not a big deal if there's a diff here I think.

File: scripts/generate-google3-docs/resource-reference/overview.md
Comment (justinsb) [inline scripts/generate-google3-docs/resource-reference/overview.md:195]: Not a problem, but why has this popped up here?  I'm confused why mockgcp would trigger this.  Did you add this manually?

File: tests/e2e/unified_test.go
Comment (justinsb) [inline tests/e2e/unified_test.go:533]: Copy/paste error? (duplicated line)

**Comment by justinsb** [comment]
A few thoughts but I don't think anything is blocking - nice work!

I'm going to approve & hold, up to you if you want to fix anything here or address in another PR (or not at all :-) )

/approve
/lgtm
/hold

File: scripts/generate-google3-docs/resource-reference/overview.md
Comment (gemmahou) [reply scripts/generate-google3-docs/resource-reference/overview.md:195]: I added this manually because the resource document is missing from our resources overview: https://screenshot.googleplex.com/4LYei5wFDfKwBxg.png

File: mockgcp/mockcompute/firewallpoliciesv1.go
Comment (gemmahou) [reply mockgcp/mockcompute/firewallpoliciesv1.go:56]: We have another usage of `id` below on line73: obj.Id = PtrTo(id)

File: pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log
Comment (gemmahou) [reply pkg/test/resourcefixture/testdata/basic/compute/v1beta1/computefirewallpolicy/_http.log]: Update the log to match realGCP

**Review by justinsb — APPROVED** [review]

**Pull request merged**
